coderd/database/models.go

```diff
@@ -1209,6 +1209,68 @@ func AllPortShareProtocolValues() []PortShareProtocol {
 	}
 }
 
+// The status of a provisioner daemon.
+type ProvisionerDaemonStatus string
+
+const (
+	ProvisionerDaemonStatusOffline ProvisionerDaemonStatus = "offline"
+	ProvisionerDaemonStatusIdle    ProvisionerDaemonStatus = "idle"
+	ProvisionerDaemonStatusBusy    ProvisionerDaemonStatus = "busy"
+)
+
+func (e *ProvisionerDaemonStatus) Scan(src interface{}) error {
+	switch s := src.(type) {
+	case []byte:
+		*e = ProvisionerDaemonStatus(s)
+	case string:
+		*e = ProvisionerDaemonStatus(s)
+	default:
+		return fmt.Errorf("unsupported scan type for ProvisionerDaemonStatus: %T", src)
+	}
+	return nil
+}
+
+type NullProvisionerDaemonStatus struct {
+	ProvisionerDaemonStatus ProvisionerDaemonStatus `json:"provisioner_daemon_status"`
+	Valid                   bool                    `json:"valid"` // Valid is true if ProvisionerDaemonStatus is not NULL
+}
```

Comment:
When will a provisioner daemon have a null status?

Reply:
Actually, you were right to flag this. Not sure where this is coming from. Perhaps the generated code doesn't erase stuff that was previously part of a query? 🤔

We currently don't have status filter for provisioner daemons. Jobs OTOH do (which I mixed your question up with).

Reply:
My gut feeling is that this is somehow being interpreted as nullable by whatever is generating this (sqlc?)